onebusaway-android/src/main/java/org/onebusaway/android/ui/ArrivalsListFragment.java

```diff
@@ -1616,6 +1621,23 @@ private void addToDB(ObaStop stop) {
         ObaContract.Stops.insertOrUpdate(stop.getId(), values, true);
     }
 
+    private void addRouteToDB(ObaArrivalInfo info) {
+        String routeId = info.getRouteId();
+        String shortName = info.getShortName();
+        String longName = info.getRouteLongName();
+        longName = ConversionUtils.getRouteLongNameSafe(longName, shortName, false);
```

Comment:
`ConversionUtils.getRouteLongNameSafe()` in the `directions` package really needs to be eliminated (it was the side-effect of a quick merge to get something out the door fast for planning trips), as it's basically redundant.  Let's tackle that in a separate issue (feel free to open).  For here, please replace with `UIUtils.getRouteDisplayName(String routeShortName, String routeLongName)`.

Reply:
Also, this might be why you're sometimes getting a blank name in the list.

Reply:
we actually need the description for it to match what's in the recents list. We can get it from `UIUtils.getRouteDescription(ObaRoute route)` but we don't have an instance of `ObaRoute`, so the "long name" is still going to be short name most of the time (until you click). But for now, I have changed it to use `getRouteDisplayName`